packages/cloud_firestore/test/cloud_firestore_test.dart

```diff
@@ -276,6 +284,26 @@ void main() {
           ]),
         );
       });
+      test('get', () async {
+        final DocumentSnapshot snapshot =
+            await collectionReference.document('bar').get();
+        expect(
+          log,
+          equals(<Matcher>[
+            isMethodCall(
+              'DocumentReference#get',
+              arguments: <String, dynamic>{'path': 'foo/bar'},
+            ),
+          ]),
+        );
+        expect(snapshot.reference.path, equals('foo/bar'));
+        expect(snapshot.data.containsKey('key1'), equals(true));
```

Comment:
This seems redundant with the line below

Reply:
In this case I'm checking that the key `key1` exists, then below I'm checking the value of the mapped to the key is `val1`. I'm happy to remove this check if it still seems unnecessary.

Reply:
There is no situation where `snapshot.data['key1']` equals `'val1'` but `snapshot.data.containsKey('key1')` is `false`. So the `containsKey` check seems redundant and I think it would be more readable to remove it. This is a minor nit, though.